network/p2p/dht/dht.go

```diff
@@ -0,0 +1,70 @@
+package dht
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+)
+
+const routingTableRefresh time.Duration = time.Second * 100
+const peerDiscoveryTimeout time.Duration = time.Second * 2
+const FlowRendezVousStr string = "flow"
+
+// THis produces a new IPFS DHT
+// on the name, see https://github.com/libp2p/go-libp2p-kad-dht/issues/337
+func NewDHT(ctx context.Context, host host.Host, bootstrapPeers []peer.AddrInfo) (*discovery.RoutingDiscovery, error) {
+	var defaultOptions = []dht.Option{
+
+		dht.ProtocolPrefix("/flow"),
+
+		dht.RoutingTableRefreshPeriod(routingTableRefresh),        // this is 100 seconds
+		dht.RoutingTableRefreshQueryTimeout(peerDiscoveryTimeout), // this is 2 seconds
+
+		// public capabilities we don't need
+		dht.DisableProviders(),
+		dht.DisableValues(),
+	}
+
+	// If we have no bootstrapPeers, we're the server
+	if len(bootstrapPeers) == 0 {
```

Comment:
Can you explain this? If there are no bootstrap peers, how does the node join the DHT network?

Reply:
It is a server that expects the other nodes to join it (and accepts them as peers). That would be the behavior of the staked ANs.

In particular, a piece of code that's not in this WIP is how the unstaked ANs are started through a constructor that passes a non-empty list of bootstrap peers here (they pass in specifically the staked ANs).

Reply:
But, the staked AN needs to know about the other staked AN's which participate in the DHT right? 

My understanding of how we were going to set this up was that the staked AN's together would form a DHT, and the values stored in this DHT would be the node information of the unstaked nodes (but perhaps this isn't accurate?).

Are we relying on the fact that the staked AN's already know about each other through bootstrap data / protocol state, and therefore don't have to explicitly connect to each other here?

Reply:
This is a demo PR, not the actual data flow.

> My understanding of how we were going to set this up was that the staked AN's together would form a DHT, and the values stored in this DHT would be the node information of the unstaked nodes (but perhaps this isn't accurate?).

Yeah, that's accurate, but just not done in this PR, which doesn't so far change the bootstrap of the node.